databricks_cli/stack/api.py

```diff
@@ -0,0 +1,353 @@
+# Databricks CLI
+# Copyright 2018 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"), except
+# that the use of services to which certain application programming
+# interfaces (each, an "API") connect requires that the user first obtain
+# a license for the use of the APIs from Databricks, Inc. ("Databricks"),
+# by creating an account at www.databricks.com and agreeing to either (a)
+# the Community Edition Terms of Service, (b) the Databricks Terms of
+# Service, or (c) another written agreement between Licensee and Databricks
+# for the use of the APIs.
+#
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import json
+from datetime import datetime
+import time
+
+import click
+
+from requests.exceptions import HTTPError
+from databricks_cli.jobs.api import JobsApi
+from databricks_cli.version import version as CLI_VERSION
+from databricks_cli.configure.config import get_profile_from_context, get_config_for_profile
+from databricks_cli.stack.exceptions import StackError
+
+DEBUG_MODE = False
+_home = os.path.expanduser('~')
+MS_SEC = 1000
+STACK_STATUS_INSERT = 'deployed'
+
+# Resource Services
+JOBS_SERVICE = 'jobs'
+
+# Config Outer Fields
+STACK_NAME = 'name'
+STACK_RESOURCES = 'resources'
+STACK_DEPLOYED = 'deployed'
+
+# Resource Fields
+RESOURCE_ID = 'id'
+RESOURCE_SERVICE = 'service'
+RESOURCE_PROPERTIES = 'properties'
+
+# Deployed Resource Fields
+RESOURCE_PHYSICAL_ID = 'physical_id'
+RESOURCE_DEPLOY_OUTPUT = 'deploy_output'
+RESOURCE_DEPLOY_TIMESTAMP = 'timestamp'
+CLI_VERSION_KEY = 'cli_version'
+
+
+class StackApi(object):
+    def __init__(self, api_client):
+        self.jobs_client = JobsApi(api_client)
+        self.host = "/"  # default host if cannot get host.
+        if click.get_current_context(silent=True):
```

Comment:
References to `click` should not exist inside this class. This will couple the class with the implementation details of our CLI. 

Instead, it is preferred that you delegate this functionality to the cli.py class. In this case, you can simply pass in `host` as a parameter.

Reply:
Agree, it may actually be too much overhead to maintain this host parameter, considering removing url and host functionality from this PR.

Reply:
If it helps to simply this PR, let's remove the URL part.